pkg/network/common/egressip.go

```diff
@@ -1008,14 +1023,23 @@ func (eit *EgressIPTracker) ReallocateEgressIPs() map[string][]string {
 	return allocation
 }
 
-func (eit *EgressIPTracker) recordMetrics() {
+func (eit *EgressIPTracker) recordMetrics(hs *osdnv1.HostSubnet) {
 	activeEgressIPCount := 0.0
 	for _, eipInfo := range eit.egressIPs {
 		if eipInfo.assignedNodeIP != "" {
 			activeEgressIPCount += 1
 		}
 	}
 	metrics.RecordEgressIPCount(activeEgressIPCount)
+	if eit.localIP != "" && eit.localIP == hs.HostIP && eit.CloudEgressIP {
+		capacity, err := eit.getCapacity(hs)
+		if err != nil {
+			klog.Errorf("Ignoring invalid HostSubnet %s for node metrics: %v", HostSubnetToString(hs), err)
+			return
+		}
+		nodemetrics.EgressIPCapacity.WithLabelValues(hs.Host).Set(float64(capacity))
+		nodemetrics.AssignedEgressIP.WithLabelValues(hs.Host).Set(float64(len(hs.EgressIPs)))
+	}
```

Comment:
You don't need to label the metric with the hostname; each openshift-sdn-node is only reporting the metric for its own HostIP, right?

Reply:
That's correct. changed it.